runtimeconfig/google/cloud/runtimeconfig/config.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Create / interact with Google Cloud RuntimeConfig configs."""
+
+from google.cloud.exceptions import NotFound
+from google.cloud.runtimeconfig._helpers import config_name_from_full_name
+from google.cloud.runtimeconfig._helpers import variable_name_from_full_name
+from google.cloud.runtimeconfig.variable import Variable
+
+
+class Config(object):
+    """A Config resource in the Cloud RuntimeConfig service.
+
+    This consists of metadata and a hierarchy of variables.
+
+    See:
+    https://cloud.google.com/deployment-manager/runtime-configurator/reference/rest/v1beta1/projects.configs
+
+    :type client: :class:`google.cloud.runtimeconfig.client.Client`
+    :param client: A client which holds credentials and project configuration
+                   for the config (which requires a project).
+
+    :type name: string
+    :param name: The name of the config.
+    """
+
+    def __init__(self, client, name):
+        self._client = client
+        self.name = name
+        self._properties = {}
+
+    def __repr__(self):
+        return '<Config: %s>' % (self.name,)
+
+    @property
+    def client(self):
+        """The client bound to this config."""
+        return self._client
+
+    @property
+    def description(self):
+        """Description of the config object.
+
+        See:
+        https://cloud.google.com/deployment-manager/runtime-configurator/reference/rest/v1beta1/projects.configs#resource-runtimeconfig
+
+        :rtype: string, or ``NoneType``
+        :returns: the description (None until set from the server).
+        """
+        return self._properties.get('description')
+
+    @property
+    def project(self):
+        """Project bound to the config.
+
+        :rtype: string
+        :returns: the project (derived from the client).
+        """
+        return self._client.project
+
+    @property
+    def full_name(self):
+        """Fully-qualified name of this variable.
+
+        Example:
+        ``projects/my-project/configs/my-config``
+
+        :rtype: string
+        :returns: The full name based on project and config names.
+
+        :raises: :class:`ValueError` if the config is missing a name.
```

Comment:
Typically `rtype/returns/raises` can go all together with no empty lines. No need to change it, just future reference.